### ISSUE TITLE
False-positive error in `require-event-prefix` for `get*` props

### ISSUE THREAD

**Issue by ai** [sveltejs/eslint-plugin-svelte#1413]
### Description

Let’s ignore `get*` props in [`svelte/require-event-prefix`](https://sveltejs.github.io/eslint-plugin-svelte/rules/require-event-prefix/) rule.

I think component structure like this is pretty popular:

```svelte
<script
  generics="Value extends { id: string }"
  lang="ts"
>
  import type { Snippet } from 'svelte'

  let {
    getHref,
    item,
    list
  }: {
    // ERROR: Component event name must start with "on". (eslint svelte/require-event-prefix)
    getHref: (value: Value) => string
    item: Snippet<[Value]>
    list: Value[]
  } = $props()
</script>

<ul>
  {#each list as i (i.id)}
    <li>
      <a href={getHref(i)}>
        {@render item(i)}
      </Clickable>
    </li>
  {/each}
</ul>
```

But `svelte/require-event-prefix` raise an error on `getHref` prop function. The rule think it could be an event listener and enforce `on` prefix.

I suggest fixing it by ignoring all props with `get*` prefix since they are definatelly not events.

**Comment by marekdedic** [comment]
Hi, I don't understand the usecase for this. Why would you pass a getter instead of the value?

**Comment by ai** [comment]
@marekdedic because this component receives a list of value and I call getter for each list item element.

**Comment by marekdedic** [comment]
Thanks, but I'm still confused... It seems to me that you could do this in one of 2 simpler ways:
- Import the getter from somewhere else
- Pass the href values directly

Would that not work?

**Comment by ai** [comment]
> Import the getter from somewhere else

Not an option since `getHref` is different

> Pass the href values directly

For my case I can generate `let links = $derived(list.map(item => [item, getHref(item)])` array, but:

1. It is a little worse in terms of performance since we will have more operations on tiny array changes.
2. It is possible that in some cases `getHref` can be called not always, so we are doing unnecessary work to generate it.

And in general throwing an error on any function passed as prop is not 100% correct since not all functions are events.

**Comment by marekdedic** [comment]
Thanks for the answers. Unfortunately, I'm still lost:
1. I don't understand why importing `getHref` is not an option
2. I don't understand the performance argument - I'm not sure passing a list of values as a prop actually causes a complete re-render (I couldn't find it in 2 mins). However, passing a getter breaks reactivity altogether, so it seems to be strictly worse?

Maybe a more fleshed-out repro is needed?

**Comment by ai** [comment]
> I don't understand why importing getHref is not an option

Because I use `Links` for different elements (`Feed` and `Posts`) and each type has own way to get href. So I can’t do href generation in the `Links` component and need to do it in `Links`’s user.

> However, passing a getter breaks reactivity altogether, so it seems to be strictly worse?

Why? I create functions in `<script>`, so the function will always have the same value.

> Maybe a more fleshed-out repro is needed?

Here are real files:
https://github.com/hplush/slowreader/blob/f6f05324ad472308bfbd23cf55492c1271dfc629/web/ui/links.svelte
https://github.com/hplush/slowreader/blob/f6f05324ad472308bfbd23cf55492c1271dfc629/web/ui/feeds.svelte
https://github.com/hplush/slowreader/blob/f6f05324ad472308bfbd23cf55492c1271dfc629/web/ui/posts.svelte

My main argument is not this case, but the main idea that we should not treat all functions in props as events.

**Comment by ota-meshi** [comment]
> My main argument is not this case, but the main idea that we should not treat all functions in props as events.

I agree with you, but I think it would be better to check whether the function type has a return type rather than excluding it with `get`.
What do you think?

---

I think there's a discussion in the thread about whether functions other than events should be passed to props, but either way, I think that check should be checked by a separate rule.

**Comment by ai** [comment]
> I agree with you, but I think it would be better to check whether the function type has a return type rather than excluding it with get.

Totally agree (but have no skills to write such rule)

**Comment by marekdedic** [comment]
Hmm, that sounds more like something discriminative. Let me think about it a little bit more and then I can make a PR
